src/main/java/toast/specialMobs/MobHelper.java

```diff
@@ -28,42 +26,37 @@
 import toast.specialMobs.entity.pigzombie.Entity_SpecialPigZombie;
 import toast.specialMobs.network.MessageExplosion;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.stream.Stream;
+
 public abstract class MobHelper
 {
+    private static Stream<EntityAITaskEntry> typeFilteredTaskEntries(EntityLiving entity, Class<?> type) {
+        return Arrays.stream(entity.tasks.taskEntries.toArray())
+            .filter(Objects::nonNull)
+            .filter(e -> e instanceof EntityAITaskEntry)
+            .map(e -> (EntityAITaskEntry) e)
+            .filter(e -> type.isInstance(e.action));
+    }
     // Clears all melee attack AIs.
     public static void clearMeleeAttackAI(EntityLiving entity) {
-        for (EntityAITaskEntry entry : (EntityAITaskEntry[]) entity.tasks.taskEntries.toArray()) if (entry.action.getClass().equals(EntityAIAttackOnCollide.class)) {
-            entity.tasks.removeTask(entry.action);
-        }
+        typeFilteredTaskEntries(entity, EntityAIAttackOnCollide.class).forEach(e -> entity.tasks.removeTask(e.action));
```

Comment:
If there are more than one matching entry this would probably result in a ConcurrentModificationException, which is probably why the original code want a toArray in the first place.

Reply:
the function does `Arrays.stream(entity.tasks.taskEntries.toArray())`, so the copy is just moved to the caller

Reply:
Ah didn't notice. Sorry for that.